content/en/faq/metrics-and-logs/mixer-migration.md

```diff
@@ -0,0 +1,16 @@
+---
+title: How do I migrate existing Mixer functionality?
+weight: 30
+---
+
+Mixer was [removed in the 1.8 Istio release](/news/releases/1.8.x/announcing-1.8/#deprecations).
+Migration is needed if you still rely on Mixer's built-in adapters or any out-of-process adapters for mesh extension.
+
+For built-in adapters, several alternatives are provided:
+
+* `Prometheus` and `Stackdriver` integrations are implemented as [proxy extensions](/docs/reference/config/proxy_extensions/).
+    Customization of telemetry generated by these two extensions can be achieved via [request classification](/docs/tasks/observability/metrics/classify-metrics/) and [Prometheus metrics customization](/docs/tasks/observability/metrics/customize-metrics/)..
+* Global and Local Rate-Limiting (`memquota` and `redisquota` adapters) functionality is provided through the [Envoy-based rate-limiting solution](/docs/tasks/policy-enforcement/rate-limit/).
+* `OPA` adapter is replaced by the [Envoy ext-authz based solution](/docs/tasks/security/authorization/authz-custom/), which supports [integration with OPA policy agent](https://www.openpolicyagent.org/docs/latest/envoy-authorization/).
+
+For custom out-of-process adapters, migration to Wasm-based extensions is strongly encouraged. Please refer to the guides on [Wasm module development](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-a-wasm-extension-with-cpp.md) and [extension distribution](/docs/ops/configuration/extensibility/wasm-module-distribution/). If you need to upgrade Istio while migrating to Wasm-based extensions, as a temporary solution, you can [enable Envoy ext-authz and gRPC access log API support in Mixer](https://github.com/istio/istio/wiki/Enabling-Envoy-Authorization-Service-and-gRPC-Access-Log-Service-With-Mixer). Note this temporary solution is not battle-tested and will unlikely get patch fixes, since it is only available on Istio 1.7 branch which is out of support window after Feb 2021.
```

Comment:
Is this Istio 1.7 piece worth mentioning?  We are towards end of Feb now.

Reply:
Or maybe this is where it is not super clear.  Are you saying users should stay on Istio 1.7 or prior to 1.7 till out of process adapters are fully migrated to Wasm-based extensions?

Reply:
Updated. The temporary solution is to allow user to upgrade to post 1.7 Istio, while still using 1.7 Mixer for adapters, so that it can buy some time for their migration.

Reply:
Thanks for the clarification @bianpengyuan !